src/main/java/com/imaee/propinq/buildings/services/implementations/BuildingService.java

```diff
@@ -0,0 +1,38 @@
+package com.imaee.propinq.buildings.services.implementations;
+
+import com.imaee.propinq.buildings.controllers.responses.BuildingResponse;
+import com.imaee.propinq.buildings.data.models.Building;
+import com.imaee.propinq.buildings.data.repositories.IBuildingRepository;
+import com.imaee.propinq.buildings.mappers.BuildingMapper;
+import com.imaee.propinq.buildings.services.interfaces.IBuildingService;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class BuildingService implements IBuildingService {
+    private final IBuildingRepository buildingRepository;
+
+    public BuildingService( IBuildingRepository buildingRepository) {
+        this.buildingRepository = buildingRepository;
+    }
+
+    @Override
+    public List<BuildingResponse> getBuildings() {
+
+        return buildingRepository.findAll()
+                .stream()
+                .map(BuildingMapper::toBuildingResponse)
+                .collect(Collectors.toList());
```

Comment:
Perfecto usando esa referencia en el `map`. Ahora, en vez de utilizar el `collect(Collectors.toList())` que es más arcaico y un poquito menos veloz, podés usar directamente el `toList()` que ya está optimizado.

```suggestion
                .stream()
                .map(BuildingMapper::toBuildingResponse)
                .toList();
```

Aclaro: para este caso, el `toList()` es correcto porque no vas a estar haciendo cambios sobre la lista que vas a devolver. Si quisieras hacer cambios luego de recolectar con `toList()`, tendrías una excepción. Esa es la pequeña pero gran diferencia entre estas estrategias de recolección.

Reply:
Perfecto, si uso .collect(Collector.toList()) es mas ineficiente pero devuelve un ArrayList, el cual es mutable y un toList(), devuelve una lista inmutable, pero es mas eficiente.